From 75af119cc59a90d5beefb5b4023980e272ae9090 Mon Sep 17 00:00:00 2001
From: b07117 <b07117@freescale.com>
Date: Fri, 24 Feb 2012 21:57:56 +0800
Subject: [PATCH 240/498] ENGR00174836 wrapper: correct Android YUV format
 mess usage

HAL_PIXEL_FORMAT_YCbCr_420_P use 3 YUV planes, same to I420
HAL_PIXEL_FORMAT_YCbCr_420_I use 1 plane with YUV interleaved

Removed glFinish to turn off GPU 3D during video playback

Signed-off-by: xianzhong <b07117@freescale.com>
---
 mx6/libgralloc_wrapper/framebuffer.cpp | 2 +-
 mx6/libgralloc_wrapper/gralloc.cpp     | 4 ++++
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/mx6/libgralloc_wrapper/framebuffer.cpp b/mx6/libgralloc_wrapper/framebuffer.cpp
index 7653329..e3be749 100755
--- a/mx6/libgralloc_wrapper/framebuffer.cpp
+++ b/mx6/libgralloc_wrapper/framebuffer.cpp
@@ -211,7 +211,7 @@ static int fb_post(struct framebuffer_device_t* dev, buffer_handle_t buffer)
 
 static int fb_compositionComplete(struct framebuffer_device_t* dev)
 {
-    glFinish();
+  //  glFinish();
     return 0;
 }
 
diff --git a/mx6/libgralloc_wrapper/gralloc.cpp b/mx6/libgralloc_wrapper/gralloc.cpp
index 7eadb9e..39ec399 100755
--- a/mx6/libgralloc_wrapper/gralloc.cpp
+++ b/mx6/libgralloc_wrapper/gralloc.cpp
@@ -118,6 +118,10 @@ static int gralloc_alloc(alloc_device_t* dev,
         return -EINVAL;
     }
 
+    //Correct android YUV 420 format mess usage by players
+    if(format == HAL_PIXEL_FORMAT_YCbCr_420_I)format = HAL_PIXEL_FORMAT_YCbCr_420_P;
+    else if(format == HAL_PIXEL_FORMAT_YCbCr_420_P)format = HAL_PIXEL_FORMAT_YCbCr_420_I;
+
     return m->gpu_device->alloc(dev, w, h, format, usage, pHandle, pStride);
 }
 
-- 
1.8.0

